### ISSUE TITLE
How to implement adb exec-out with adbkit ?

### ISSUE THREAD

**Issue by phurinaix** [DeviceFarmer/adbkit#258]
to get the UI elements (In XML format) you can achieve this by using this command
```adb exec-out uiautomator dump /dev/tty```

I use ```device.shell('exec-out uiautomator dump /dev/tty')```from adbkit, But it doesn't work.

So I used child_process instead (native library in Node.js) to call this command.

```
const { spawn } = require("child_process");

const spawnChild = async (command: string, args: Array<string>) => {
  const child = spawn(command, args);

  let data = "";

  for await (const chunk of child.stdout) {
    data += chunk;
  }
  let error = "";
  for await (const chunk of child.stderr) {
    error += chunk;
  }
  const exitCode = await new Promise((resolve, reject) => {
    child.on("close", resolve);
  });

  if (exitCode) {
    throw new Error(`subprocess error exit ${exitCode}, ${error}`);
  }
  return data;
};

spawnChild("adb", ["exec-out", "uiautomator", "dump", "/dev/tty"]).then(result => console.log(result))

```

I want to know how to implement adb exec-out with adbkit library

Or is there any method or API from adbkit to get the UI elements of an Android device ?

**Comment by UrielCh** [comment]
```
const testUiautomator = async (deviceClient: DeviceClient) => {
  const duplex = await deviceClient.shell('uiautomator dump /dev/tty');
  const result = await new Parser(duplex).readAll();
  console.log(result.toString('utf8'));
}
```

tested here:

https://github.com/UrielCh/adbkit/blob/master/tasks/realTest.ts

**Comment by UrielCh** [comment]
exec(command: string | ArrayLike<WithToString>): Promise<Duplex>;
execOut(command: string | ArrayLike<WithToString>): Promise<Buffer>;
execOut(command: string | ArrayLike<WithToString>, encoding: BufferEncoding): Promise<string>;

Will be available in `u4/adbkit@3.1.5` within days.

the adb `exec:` was not implemented in the project, all commands used to relay on `shell:`

**Comment by phurinaix** [comment]
thanks
